site/app/controllers/banner/BannerController.php

```diff
@@ -0,0 +1,225 @@
+     *
+     * @return WebResponse
+     * @see GlobalController::prep_user_sidebar
+     * @see BannerView::showEventBanners
+     */
+    public function viewCommunityEvents(): WebResponse {
+        $communityEventBanners = $this->core->getSubmittyEntityManager()->getRepository(BannerImage::class) ->findAll();
+        return new WebResponse(BannerView::class, 'showEventBanners', $communityEventBanners);
+    }
+
+
+
+    /**
+     * @Route("/banner/upload", methods={"POST"})
+     */
+    public function ajaxUploadEventFiles(): JsonResponse {
+        $upload_path = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "community_events");
+
+        if (isset($_POST['release_time'])) {
+            $release_date = DateUtils::parseDateTime($_POST['release_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+
+
+        if (isset($_POST['close_time'])) {
+            $close_date = DateUtils::parseDateTime($_POST['close_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+        if (!isset($_FILES["files1"]) || $_FILES["files1"] === []) {
+            return JsonResponse::getErrorResponse("No files were submitted.");
+        }
+
+        $uploaded_files = $_FILES["files1"];
+        $count_item = count($uploaded_files["name"]);
+        $extra_name = $_POST['extra_name'];
+        $link_name = $_POST['link_name'];
+
+        if ($extra_name == ".." || $link_name == "..") {
+            return JsonResponse::getErrorResponse("invalid name");
+        }
+
+        if ($extra_name === "") {
+            if ($count_item !== 1) {
+                return JsonResponse::getErrorResponse("You can only have one banner submitted.");
+            }
+        }
+        else {
+            if ($count_item > 2) {
+                return JsonResponse::getErrorResponse("Can't have more than two banners submitted.");
+            }
+        }
+
+
+        $specificPath = $close_date->format("Y");
+        $actual_banner_name = "";
+
+        for ($j = 0; $j < $count_item; $j++) {
+            if ($uploaded_files['name'][$j] == "..") {
+                return JsonResponse::getErrorResponse("invalid name");
+            }
+
+            if ($uploaded_files['name'][$j] != $extra_name) {
+                $actual_banner_name = $uploaded_files['name'][$j];
+            }
+        }
+
+        $currentDate = new \DateTime();
+        $folder_made_name = $actual_banner_name . "Folder" . $currentDate->format('Y-m-d_H-i-s');
+
+        $full_path = FileUtils::joinPaths($upload_path, $specificPath, $folder_made_name);
+
+        // if (!$this->core->getAccess()->canI("path.write", ["dir" => "community_events", "path" => $full_path])) {
+        //     return JsonResponse::getErrorResponse("Bad File Request");
+        // }
```

Comment:
Were you not able to get these working?

Reply:
yea I tested them out again and I keep getting that I have a bad read.